lib/pkcs11h-certificate.c

```diff
@@ -786,9 +786,6 @@ __pkcs11h_certificate_doPrivateOperation (
 #if defined(ENABLE_PKCS11H_THREADING)
 	PKCS11H_BOOL mutex_locked = FALSE;
 #endif
-	CK_MECHANISM mech = {
-		mech_type, NULL, 0
-	};
 
```

Comment:
can you please add assert with NULL for all these?

can you please add a debug with mech_type after the asserts so that we have some visibility of what is going on?